scaladoc/test/dotty/tools/scaladoc/snippets/SnippetsE2eTest.scala

```diff
@@ -0,0 +1,138 @@
+    assertTrue(report(s"Invalid outer line offset: ${ws.outerLineOffset}. Expected: ${si.outerOffset.line}\n"), ws.outerLineOffset == si.outerOffset.line)
+    assertTrue(report(s"Invalid outer column offset: ${ws.outerColumnOffset}. Expected: ${si.outerOffset.column}\n"), ws.outerColumnOffset == si.outerOffset.column)
+    assertTrue(report(s"Invalid inner line offset: ${ws.innerLineOffset}. Expected: ${si.innerOffset.line}\n"), ws.innerLineOffset == si.innerOffset.line)
+    assertTrue(report(s"Invalid inner column offset: ${ws.innerColumnOffset}. Expected: ${si.innerOffset.column}\n"), ws.innerColumnOffset == si.innerOffset.column)
+  }
+
+  private def checkMessages(compilationMessages: Seq[SnippetCompilerMessage], messages: Seq[Message], ws: WrappedSnippet) = {
+    val compilationMessagesWithPos = compilationMessages.collect {
+      case m @ SnippetCompilerMessage(Some(_), _, _) => m
+    }.toList
+    val result = messages.flatMap { msg =>
+      compilationMessagesWithPos.find { cmsg =>
+        cmsg.level == msg.level && cmsg.position.get.line == msg.offset.line && cmsg.position.get.column == msg.offset.column
+      }.fold(Seq(s"Expected ${msg.level.text} message at ${msg.offset.line}:${msg.offset.column}.")) { resp =>
+        val pos = resp.position.get
+        if debug then {
+          if !(pos.relativeLine == pos.line - ws.outerLineOffset + ws.innerLineOffset) then Seq(
+            s"Expected ${msg.level.text} message at relative line: ${pos.line - ws.outerLineOffset + ws.innerLineOffset} but found at ${pos.relativeLine}"
+          ) else Nil
+        } else {
+          if !(pos.relativeLine == pos.line - ws.outerLineOffset) then Seq(
+            s"Expected ${msg.level.text} message at relative line: ${pos.line - ws.outerLineOffset} but found at ${pos.relativeLine}"
+          ) else Nil
+        }
+      }
+    }
+    if !result.isEmpty then {
+      val errors = result.mkString("\n")
+      val foundMessages = compilationMessages.map(m => s"${m.level} at ${m.position.get.line}:${m.position.get.column}").mkString("\n")
+      throw AssertionError(Seq("Errors:", errors,"Found:", foundMessages).mkString("\n", "\n", "\n"))
+    }
+  }
+
+  def moduleTestingFunc: DocContext ?=> Module => Unit = (m: Module) => {
+    val snippets = m.members.values
+      .flatMap(_.docs)
+      .map(_.body)
+      .collect { case n: mdu.Node => n }
+      .flatMap(_.getDescendants.asScala)
+      .collect { case en: ExtendedFencedCodeBlock => en }
+
+    assertTrue(report(s"Expected $snippetsCount snippets but found ${snippets.size}"), snippets.size == snippetsCount)
+
+    snippets.foreach { snippet =>
+      val configStrs = (snippet.getPrevious() match {
+        case c: mdu.ContentNode =>
+          c.getContentChars.toString.split("\n").map(_.trim)
+        case _ => throw AssertionError(s"Not found info for snippet ${snippet.codeBlock.getContentChars.toString}")
+      }).toList
+      val info = SnippetInfo(configStrs.head)
```

Comment:
`head` should work on others collections as well (so `toList` may not be needed)

Reply:
.split method is defined in Java and returns Array[String], thats why we need to call .toList